pyrovision/datasets/openfire.py

```diff
@@ -4,195 +4,69 @@
 # See LICENSE or go to <https://www.gnu.org/licenses/agpl-3.0.txt> for full license details.
 
 from pathlib import Path
-import warnings
-import json
-from PIL import Image, ImageFile
+from torchvision.datasets import DatasetFolder
+from typing import Any, Callable, Optional
+from PIL import Image
+from .utils import download_dataset_from_url
 
-from torchvision.datasets import VisionDataset
-from .utils import download_url, download_urls
+__all__ = ['OpenFire']
 
-ImageFile.LOAD_TRUNCATED_IMAGES = True
 
-__all__ = ['OpenFire']
+url = "https://github.com/pyronear/pyro-vision/releases/download/v0.1.2/open_fire-dd0b5bb0.zip"
+url_sample = "https://github.com/pyronear/pyro-vision/releases/download/v0.1.2/open_fire_sample-06a07e48.zip"
+zip_filename = "open_fire.zip"
+
 
+def pil_loader(path: str) -> Image.Image:
+    # open path as file to avoid ResourceWarning (https://github.com/python-pillow/Pillow/issues/835)
+    with open(path, 'rb') as f:
+        img = Image.open(f)
+        return img.convert('RGB')
 
-class OpenFire(VisionDataset):
+
+class OpenFire(DatasetFolder):
```

Comment:
In the process of updating the class, several methods were removed. Are they covered by DatasetFolder? (len, getitem, etc.)

Reply:
yep : https://github.com/pytorch/vision/blob/730c5e1eab130e2900c8e839ea08fa11f024516f/torchvision/datasets/folder.py#L223